### PR TITLE
recording: record the snaps installed on the machine

### PR DESCRIPTION
**Pull request by elopio**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----
This requires #1566

### CODE REVIEW

File: snapcraft/tests/fixture_setup.py [hunk 1]
```diff
@@ -595,47 +598,6 @@ def terminate(self):
         return Popen(args)
 
 
-class FakeSnapd(fixtures.Fixture):
```
Comment (kalikiana) [inline snapcraft/tests/fixture_setup.py]: Would you mind leaving the FakeSnapd classes where they are and do this in a separate branch (that ideally actually merges them)? Here it adds a lot of noise.

Comment (elopio) [reply snapcraft/tests/fixture_setup.py]: I'm trying to merge them in a follow up branch, and the easiest step to get there is to move your fake to the lxd test module.

You can take a look at what I'm trying to do here:
https://github.com/elopio/snapcraft/commit/240772d0806b04def52c23a3022172cac85ba08a

**Comment by elopio** [comment]
@kalikiana snapd fake merge proposal ready in #1569.

File: snapcraft/tests/fixture_setup.py (hunk 1 above)
Comment (kalikiana) [reply snapcraft/tests/fixture_setup.py]: I don't see the reasoning... but that PR looks lovely, so I'm happy to accomodate.

File: snapcraft/tests/test_lifecycle.py [hunk 2]
```diff
@@ -757,6 +762,48 @@ def test_prime_with_build_info_records_manifest(self):
...
+             'revision': 'test-snap-2-revision'},
+        ]
+
+        self.make_snapcraft_yaml("""parts:
+  test-part:
+    plugin: nil
+""")
+        lifecycle.execute('prime', self.project_options)
+
+        expected = ("""name: test
```
[the remark this answers is not in the document]
Comment (elopio) [reply snapcraft/tests/test_lifecycle.py]: Many non-related tests in this file required dedent. So I proposed it in #1584.

**Pull request merged**